src/main/java/com/example/rentabookrestservices/service/BookService.java

```diff
@@ -0,0 +1,69 @@
+package com.example.rentabookrestservices.service;
+
+import com.example.rentabookrestservices.domain.Book;
+import com.example.rentabookrestservices.domain.BookPrice;
+import com.example.rentabookrestservices.dto.BookCreateDto;
+import com.example.rentabookrestservices.exception.BookNotFoundException;
+import com.example.rentabookrestservices.mapper.BookMapper;
+import com.example.rentabookrestservices.repository.BookPriceRepository;
+import com.example.rentabookrestservices.repository.BookRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+public class BookService {
+    private final BookRepository bookRepository;
+    private final BookPriceRepository bookPriceRepository;
+
+    public BookService(BookRepository bookRepository, BookPriceRepository bookPriceRepository) {
+        this.bookRepository = bookRepository;
+        this.bookPriceRepository = bookPriceRepository;
+    }
+
+    public List<Book> getAllBooks() {
+        return bookRepository.findAll();
+    }
+
+    public Book getBookById(long id) {
+        return bookRepository.findById(id)
+                .orElseThrow(() -> new BookNotFoundException(id));
+    }
+
+    public ResponseEntity<Book> createBook(BookCreateDto bookCreateDto) {
+        Book book = BookMapper.INSTANCE.bookCreateDtoToBook(bookCreateDto);
+        BookPrice bookPrice = BookMapper.INSTANCE.bookPriceCreateDtoToBookPrice(bookCreateDto.getBookPriceCreateDto());
+
+        bookPrice.setStartDate(LocalDate.now());
+        bookPrice.setEndDate(LocalDate.parse("9999-12-31"));
+        bookPriceRepository.save(bookPrice);
+        book.setBookPrice(bookPrice);
+        return new ResponseEntity<>(bookRepository.save(book), HttpStatus.CREATED);
+    }
+
+    public Book updateBook(long id, Book _book) {
+        Book book = bookRepository.findById(id).orElseThrow(() -> new BookNotFoundException(id));
+        BookPrice bookPrice = bookPriceRepository
+                .findById(_book.getBookPrice().getId()).orElseThrow();
+
+        book.setIsbn(_book.getIsbn());
+        book.setName(_book.getName());
+        book.setPages(_book.getPages());
+        book.setPublishYear(_book.getPublishYear());
+
+        bookPrice.setPrice(_book.getBookPrice().getPrice());
+        bookPrice.setStartDate(_book.getBookPrice().getStartDate());
+        bookPrice.setEndDate(_book.getBookPrice().getEndDate());
+
+        bookPriceRepository.save(bookPrice);
+        return bookRepository.save(book);
+    }
+
+    public ResponseEntity<HttpStatus> deleteBook(Long id) {
```

Comment:
bu method boolean dönebilir. `ResponseEntity<HttpStatus>` dönüşünü REST katmanında ilgili http status kodu ile sağlarız. HTTP status kodu gibi REST katmanını ilgilendiren işlemleri business service sınıfına taşımamış oluruz

Reply:
abi bunları dediğim gibi yüzde 50 sini bitirdim, geri kalanları da yarına tamalarım inşallah